src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -166,30 +188,7 @@ internal HttpConnection(HttpConnectionOptions httpConnectionOptions, ILoggerFact
         /// A connection cannot be restarted after it has stopped. To restart a connection
         /// a new instance should be created using the same options.
         /// </remarks>
-        public Task StartAsync(CancellationToken cancellationToken = default)
-        {
-            return StartAsync(TransferFormat.Binary, cancellationToken);
-        }
-
-        /// <summary>
-        /// Starts the connection using the specified transfer format.
-        /// </summary>
-        /// <param name="transferFormat">The transfer format the connection should use.</param>
-        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
-        /// <returns>A <see cref="Task"/> that represents the asynchronous start.</returns>
-        /// <remarks>
-        /// A connection cannot be restarted after it has stopped. To restart a connection
-        /// a new instance should be created using the same options.
-        /// </remarks>
-        public async Task StartAsync(TransferFormat transferFormat, CancellationToken cancellationToken = default)
```

Comment:
Keep this constructor. It can override the one passed in the constructor. That would avoid breaking the Azure SignalR SDK:

https://github.com/Azure/azure-signalr/blob/7257fda52f219601f8efc34dd240728a9c4ea31c/src/Microsoft.Azure.SignalR.Common/ServiceConnections/ConnectionFactory.cs#L32-L44

Reply:
Dang it. This isn't the constructor though. This change is easier if we can lock

Reply:
We can probably keep that working. Do you wanted to make HttpConnectionFactory internal, but keep HttpConnection public?

Reply:
Isn't the Azure SignalR SDK going to have to change their Connection factory to implement the new IConnectionFactory interface anyway? Might as well break to the cleaner API.

Reply:
Isn't the Azure SignalR SDK going to have to change their Connection factory to implement the new IConnectionFactory interface anyway? Might as well break to the cleaner API.

Reply:
They don't have to because they don't use the HubConnection.

Reply:
They must be implementing the old IConnectionFactory for a reason though. The only reason I could think of is so it could be used by HubConnection. Unless you plan on keeping two IConnectionFactory's around in 3.0 and having HubConnection support both, the Azure SignalR SDK is going to have to update their ConnectionFactory implementation, so I don't see the value in keeping this legacy StartAsync overload around for the Azure SignalR SDK specifically.

Reply:
They’re not implementing the old factory. They actually have their own factory (it’s just a copy of the shape from SignalR). So none of this will affect them (other than maybe a potential type conflict). I know because I originally wrote that code